### ISSUE TITLE
About the Emu Edit Benchmark metrics.

### ISSUE THREAD

**Issue by hanzhn** [HaozheZhao/UltraEdit#18]
I appreciate your excellent work of Instruction-based Editing. Thanks for your efforts!

I have some questions for you about the Emu Edit Benchmark metrics.

1. What is the specific version of CLIP and DINO to calculate these metrics? I can't find any clue in Emu Edit and yours report.
2. Have you noticed that the dataset Emu Edit provided in HuggingFace repo: [emu_edit_test_set_generations](https://huggingface.co/datasets/facebook/emu_edit_test_set_generations) may have mistakenly swapped the test and validation set, of which the record numbers are not aligned with the paper and their another HuggingFace repo: [emu_edit_test_set](https://huggingface.co/datasets/facebook/emu_edit_test_set). If this is true, which dataset split should I use for the metrics calculation, and which split did you use for your reported numbers?

**Comment by hanzhn** [comment]
Or can you point out where I can find authoritative code for these calculation? That will be helpful.

**Comment by HaozheZhao** [comment]
Hi,

Thank you for bringing these issues to our attention.

### Versioning for Metrics Calculation

We've noticed that the original Emu Edit paper and dataset do not specify the versions of CLIP and DINO used. To align with other benchmarks, we adopted the settings used by MagicBrush ([GitHub Repository](https://github.com/OSU-NLP-Group/MagicBrush)). Specifically, the versions are `"ViT-B/32"` for CLIP and `"dino_vits16"` for DINO embeddings. We ensured consistency by rerunning all results in our paper based on the Emu Edit benchmark.

### Dataset Splits and Inconsistencies

Regarding the dataset split issue: we utilized the test set of [emu_edit_test_set](https://huggingface.co/datasets/facebook/emu_edit_test_set) for our evaluations. And due to mistakenly swapped dataset, our reported results were based on the validation set from the [emu_edit_test_set_generations](https://huggingface.co/datasets/facebook/emu_edit_test_set_generations). 

Also, there are known issues with the benchmark quality as discussed in [this discussion thread](https://huggingface.co/datasets/facebook/emu_edit_test_set_generations/discussions/3). Some image-caption pairs seem incorrect, like placeholder captions (e.g., 'a train station in city') or identical source and target captions.

### Evaluation Code

For the metrics evaluation, we adhered closely to the MagicBrush evaluation script ([GitHub Link](https://github.com/OSU-NLP-Group/MagicBrush/blob/main/evaluation/image_eval.py)) for both benchmarks with no major modifications. We plan to share our refined evaluation code soon; however, in the meantime, you can refer to the provided script in MagicBrush for immediate use.

**Comment by Eureka-Maggie** [comment]
So in response to the Emu Edit benchmark issues, how did you apply this test set to your testing? In particular, can you tell me how the clip_dir was calculated?

**Comment by llCurious** [comment]
hi @HaozheZhao . Nice work.
I noticed that in  [clip_similarity.py](https://github.com/HaozheZhao/UltraEdit/blob/main/data_generation/metrics/clip_similarity.py#L22-L23), you used `dinov2_vitl14_reg` rather than `dino_vits16`. 

I am wondering which one is actually used in the evaluation? BTW, the clip model is also different `ViT-B/32 ` or `ViT-L/14`？

**Comment by HaozheZhao** [comment]
> hi [@HaozheZhao](https://github.com/HaozheZhao) . Nice work. I noticed that in [clip_similarity.py](https://github.com/HaozheZhao/UltraEdit/blob/main/data_generation/metrics/clip_similarity.py#L22-L23), you used `dinov2_vitl14_reg` rather than `dino_vits16`.
> 
> I am wondering which one is actually used in the evaluation? BTW, the clip model is also different `ViT-B/32 ` or `ViT-L/14`？

Hi @llCurious,  

Thank you for your question!  

The [`clip_similarity.py`](https://github.com/HaozheZhao/UltraEdit/blob/main/data_generation/metrics/clip_similarity.py#L22-L23) you mentioned is only used to compute statistics for the generated data as part of the data generation pipeline. It is **not** the evaluation metric used in our experiments.  

For the experiments, we strictly adhere to the model settings of MagicBrush, using `dino_vits16` and `ViT-B/32`. You can refer to the [MagicBrush evaluation script](https://github.com/OSU-NLP-Group/MagicBrush/blob/main/evaluation/image_eval.py) for more details.  

Let me know if you have further questions!

**Comment by HaozheZhao** [comment]
Hi @Eureka-Maggie,

You can find more details in Appendix C.2 of our paper. The CLIP_dir is calculated using the code from the [MagicBrush evaluation script](https://github.com/OSU-NLP-Group/MagicBrush/blob/main/evaluation/image_eval.py).

To summarize:
We use the CLIP text-image similarity between the edited image and the output caption, as well as the CLIP text-image direction similarity (CLIP_dir), to measure the model's instruction-following ability. Specifically, CLIP_dir evaluates the agreement between the change in caption embeddings and the change in image embeddings.

Since the EmuEdit paper does not specify the versions of the CLIP and DINO models used for metrics, we adopted the settings from MagicBrush to maintain consistency with other benchmarks. Specifically, we use ViT-B/32 for CLIP and dino_vits16 for DINO embeddings. To ensure fair and consistent evaluation, we reran all baseline results on the EmuEdit benchmark using these settings.

Additionally, we observed certain issues with the quality of the EmuEdit benchmark, including incorrect image-caption pairs, placeholder captions (e.g., "a train station in city"), or cases where the source and target captions are identical. To address these issues, we removed such incorrect cases prior to evaluation.

Let me know if you have further questions!

**Comment by River-Zhang** [comment]
Thank you very much for your kind response!

We plan to follow the testing settings outlined in your paper to evaluate our own model and use the benchmark results as a reference. I was wondering if you could kindly provide the list of incorrect case IDs that were removed from the EMU edit set, so that we can replicate the same setup.

Additionally, I would like to confirm that the EMU model results in your paper were tested on the corrected EMU dataset, correct?

Thank you again for your assistance, and I look forward to hearing from you.

**Comment by rahatkutubi** [comment]
Hi

 Emu Edit test has no ground truth images,  how do they calculate DINO score, L1 score? Is it between the input image and the edited images?

**Comment by HaozheZhao** [comment]
> Hi
> 
> Emu Edit test has no ground truth images, how do they calculate DINO score, L1 score? Is it between the input image and the edited images?

Yes! You are right about this. They do calculate these score between the input image and the image your model generated. As for text-image similiarity, the EMU Edit benchmark calculated the score between the generated images and the output caption they provided in the benchmark.
